sdk/communication/azure-communication-chat/azure/communication/chat/_communication_identifier_serializer.py

```diff
@@ -3,130 +3,103 @@
-        :param identifierModel: CommunicationIdentifierModel
-        :type identifierModel: CommunicationIdentifierModel
-        :return: Union[CommunicationUserIdentifier, CommunicationPhoneNumberIdentifier]
-        :rtype: Union[CommunicationUserIdentifier, CommunicationPhoneNumberIdentifier]
-        :rasies: ValueError
-        """
-
-        raw_id = identifierModel.raw_id
-        if not raw_id:
-            raise ValueError("Identifier must have a valid id")
-
-        CommunicationUserIdentifierSerializer.assertMaximumOneNestedModel(identifierModel)
-
-        if identifierModel.communication_user is not None:
-            return CommunicationUserIdentifier(raw_id)
-        if identifierModel.phone_number is not None:
-            if not identifierModel.phone_number:
-                raise ValueError("PhoneNumberIdentifier must have a valid attribute - phone_number")
-            return PhoneNumberIdentifier(identifierModel.phone_number.value, raw_id=raw_id)
-        if identifierModel.microsoft_teams_user is not None:
-            if identifierModel.microsoft_teams_user.is_anonymous not in [True, False]:
-                raise ValueError("MicrosoftTeamsUser must have a valid attribute - is_anonymous")
-            if not identifierModel.microsoft_teams_user.user_id:
-                raise ValueError("MicrosoftTeamsUser must have a valid attribute - user_id")
-            if not identifierModel.microsoft_teams_user.cloud:
-                raise ValueError("MicrosoftTeamsUser must have a valid attribute - cloud")
-            return MicrosoftTeamsUserIdentifier(
-                raw_id=raw_id,
-                user_id=identifierModel.microsoft_teams_user.user_id,
-                is_anonymous=identifierModel.microsoft_teams_user.is_anonymous,
-                cloud=identifierModel.microsoft_teams_user.cloud
-            )
-
-        return UnknownIdentifier(raw_id)
-
-    @classmethod
-    def _getIdentifierType(cls, communicationIdentifier):
-        def has_attributes(obj, attributes):
-            return all([hasattr(obj, attr) for attr in attributes])
-
-        if has_attributes(communicationIdentifier, ["identifier"]):
-            return _IdentifierType.COMMUNICATION_USER_IDENTIFIER
-
-        if has_attributes(communicationIdentifier, ['phone_number', 'raw_id']):
-            return _IdentifierType.PHONE_NUMBER_IDENTIFIER
-
-        if has_attributes(communicationIdentifier, ["raw_id", "user_id", "is_anonymous", "cloud"]):
-            return _IdentifierType.MICROSOFT_TEAMS_IDENTIFIER
-
-        if has_attributes(communicationIdentifier, ["raw_id"]):
-            return _IdentifierType.UNKNOWN_IDENTIFIER
+        return CommunicationIdentifierModel(
+            raw_id=identifier.id
+        )
+    except AttributeError:
+        raise TypeError("Unsupported identifier type " + identifier.__class__.__name__)
+
+
+def deserialize_identifier(identifier_model):
```

Comment:
In other sdks (.NET i.e), the serialization/deserialization logic is auto generated from the swagger (autorest). 
I guess this needs to be hand-crafted in python ? 
Would not make sense to have this in a common project so other modalities ca reuse it ?

Reply:
I didn't change this - just refactored it a bit.
Currently the response deserialization _is_ taking place in the autorest generated code - however the object returned (as represented by the swagger) is not the object we want users to interact with (as it's somewhat unwieldy) - so we then further deserialize that single object into these discrete objects.